modules/siddhi-core/src/main/java/org/wso2/siddhi/core/query/processor/stream/window/ExternalTimeBatchWindowProcessor.java

```diff
@@ -147,28 +161,48 @@ private void processWithStartTime(ComplexEventChunk<StreamEvent> streamEventChun
         }
 
         while (streamEventChunk.hasNext()) {
+
             StreamEvent currStreamEvent = streamEventChunk.next();
-            if (currStreamEvent.getType() != ComplexEvent.Type.CURRENT) {
+
+            if (currStreamEvent.getType() == ComplexEvent.Type.TIMER) {
+
+                if (lastScheduledTime <= currStreamEvent.getTimestamp()) {
+                    // implies that there have not been any more events after this schedule has been done.
+                    flushCurrentChunk(nextProcessor, streamEventCloner, lastCurrentEventTime);
+
+                    // rescheduling to emit the current batch after expiring it if no further events arrive.
+                    lastScheduledTime = executionPlanContext.getTimestampGenerator().currentTime() + schedulerTimeout;
+                    lastCurrentEventTime = lastCurrentEventTime + timeToKeep;
+                    scheduler.notifyAt(lastScheduledTime);
```

Comment:
I think we should not have a scheduler.notifyAt(lastScheduledTime); here as it is redundant